scalding-commons/src/main/scala/com/twitter/scalding/commons/source/BinaryConverters.scala

```diff
@@ -0,0 +1,46 @@
+/*
+Copyright 2015 Twitter, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package com.twitter.scalding.commons.source
+
+import com.twitter.elephantbird.mapreduce.io.BinaryConverter
+import com.twitter.scrooge.{ BinaryThriftStructSerializer, ThriftStructCodec, ThriftStruct }
+import scala.reflect.ClassTag
+
+/*
+ * Common BinaryConverters to be used with GenericSource / GenericScheme.
+ */
+
+case object IdentityBinaryConverter extends BinaryConverter[Array[Byte]] {
+  override def fromBytes(messageBuffer: Array[Byte]) = messageBuffer
+  override def toBytes(message: Array[Byte]) = message
+}
+
+object ScroogeBinaryConverter {
+  def apply[T <: ThriftStruct: ClassTag]: BinaryConverter[T] = {
+    val ct = implicitly[ClassTag[T]]
+    new BinaryConverter[T] {
+      val serializer = BinaryThriftStructSerializer[T] {
+        val companionClass = Class.forName(ct.runtimeClass.getName + "$")
```

Comment:
could have re-used that code if the codec methods were public. What do you think? Or maybe add support for BinaryThriftStructSerializer there.

Not sure if we're planning a chill release. I can copy over the code for union here for now.